WcaOnRails/app/controllers/persons_controller.rb

```diff
@@ -25,4 +25,17 @@ def index
       end
     end
   end
+
+  def show
+    @person = Person.current.includes(:user, :ranksSingle, :ranksAverage, :competitions).find_by_wca_id(params[:id])
+    @previous_persons = Person.where(wca_id: params[:id]).where.not(subId: 1).order(:subId)
+    @ranks_single = @person.ranksSingle
+    @ranks_average = @person.ranksAverage
+    @events_competed_in = Event.where(id: (@ranks_single.map(&:eventId) + @ranks_average.map(&:eventId)).uniq)
```

Comment:
I've just realised that there's no way for `RanksAverage` object to exist without corresponding `RanksSingle` object, so I'll simply iterate over `@ranks_single` =)